roles/openshift_facts/library/openshift_facts.py

```diff
@@ -239,49 +272,56 @@ def walk_metadata(self, metadata_url, headers=None, expect_json=False):
         for line in self.query_metadata(metadata_url, headers, expect_json):
             if line.endswith('/') and not line == 'public-keys/':
                 key = line[:-1]
-                metadata[key]=self.walk_metadata(metadata_url + line, headers,
-                                                 expect_json)
+                metadata[key] = self.walk_metadata(metadata_url + line,
+                                                   headers, expect_json)
             else:
                 results = self.query_metadata(metadata_url + line, headers,
                                               expect_json)
                 if len(results) == 1:
+                    # disable pylint maybe-no-member because overloaded use of
+                    # the module name causes pylint to not detect that results
+                    # is an array or hash
+                    # pylint: disable=maybe-no-member
                     metadata[line] = results.pop()
                 else:
                     metadata[line] = results
         return metadata
 
     def get_provider_metadata(self, metadata_url, supports_recursive=False,
-                          headers=None, expect_json=False):
+                              headers=None, expect_json=False):
         try:
             if supports_recursive:
-                metadata = self.query_metadata(metadata_url, headers, expect_json)
+                metadata = self.query_metadata(metadata_url, headers,
+                                               expect_json)
             else:
-                metadata = self.walk_metadata(metadata_url, headers, expect_json)
-        except OpenShiftFactsMetadataUnavailableError as e:
+                metadata = self.walk_metadata(metadata_url, headers,
+                                              expect_json)
+        except OpenShiftFactsMetadataUnavailableError:
             metadata = None
         return metadata
 
-    def get_ansible_facts(self):
-        if not hasattr(self, 'ansible_facts'):
-            self.ansible_facts = ansible_facts(module)
-        return self.ansible_facts
-
+    # TODO: refactor to reduce the size of this method, potentially create
+    # sub-methods (or classes for the different providers)
+    # temporarily disable pylint too-many-statements
+    # pylint: disable=too-many-statements
```

Comment:
@twiest completely understand. I'll work up a fix for this specific disable. As well as localize the disables set at the top of the script.

We are hamstrung on a few of the disables based on design decisions from ansible (which I've tried to disable these at the block level or as locally as possible).

Most of the others are to prevent having to majorly refactor the scripts,. having to address a long laundry list of stylistic pylint warnings/refactor messages alongside a bugfix PR is painful in and of itself (not to mention quite time consuming). Having to address pylint failures that require invasive refactoring of the logic of the script then introduces a larger burden of having to test those changes against multiple deployment types and against multiple providers.

Each time I hit these pylint errors for existing scripts, I try to balance the time involved to address the errors against the severity of the failure message and the potential for the fix to introduce additional bugs.

@twiest @kwoodson As an additional aside, I think we should probably have a wider discussion as a group over which pylint checks we should actively enforce vs ignore. (I'd personally see us not be so strict on line length, the presence of TODO comments, and some of the refactor failures). I think a larger discussion around pros/cons of which warnings/refactor messages should be enforced would be a better approach to balancing the usefulness of pylint vs some of the more stylistic related errors.